src/haxeLanguageServer/features/haxe/DiagnosticsFeature.hx

```diff
@@ -45,6 +46,9 @@ class DiagnosticsFeature {
 		errorUri = new FsPath(Path.join([context.workspacePath.toString(), "Error"])).toUri();
 
 		useJsonRpc = context.haxeServer.supports(DisplayMethods.Diagnostics);
+		if (context.config.user.useLegacyDiagnostics && context.haxeServer.haxeVersion < new SemVer(5, 0, 0)) {
```

Comment:
Hmm. Might be useful to skip the haxeVersion check for now and only add it when we remove legacy diagnostics support? That way people could opt out of Json RPC diagnostics with current nightlies / haxe 5 preview.

Unless this check only match for non pre release Haxe 5 versions?

Reply:
ok, I'll remove it and we should aim for a vshaxe release as quickly as possible.
once HaxeFoundation/haxe#11413 is merged we will add the check, do a new release and then the option only works on 4.3.5+ versions (old nightlies will then be forced to Json RPC if they report support for it with no opt-out).